tests/unittests/test_updated_bookmark.py

```diff
@@ -37,3 +37,53 @@ def test_update_bookmark(self, mock_now, mock_get_query_params, mock_update_book
         customers = list(CUSTOMER_OBJECT.get_objects())
         # Verify that the min-max evaluation returns correct results and provided in the update_bookmark()
         mock_update_bookmark.assert_called_with(strftime(strptime_to_utc(NOW_TIME) - datetime.timedelta(DATE_WINDOW_SIZE)))
+
+
+    @mock.patch("tap_shopify.streams.base.Stream.call_api")
+    @mock.patch("tap_shopify.streams.base.Stream.get_bookmark", return_value=strptime_to_utc('2021-08-11T01:55:05-04:00'))
+    @mock.patch("tap_shopify.streams.base.Stream.update_bookmark")
+    @mock.patch("tap_shopify.streams.base.Stream.get_query_params")
+    @mock.patch("singer.utils.now", return_value=strptime_to_utc(NOW_TIME))
+    def test_since_id_and_updated_at_max_deleted(self, mock_now, mock_get_query_params, mock_update_bookmark, mock_get_bookmark, mock_call_api):
+        """Verify after successful sync since_id and updated_at_max keys are deleted from the state"""
+        mock_call_api.return_value = [CUSTOMER_1, CUSTOMER_2]
+        Context.state = {"bookmarks": {
+            "currently_sync_stream": 'customers',
+            "customers": {
+                "updated_at": "2021-03-27T00:00:00.000000Z",
+                "since_id": 15,
+                "updated_at_max": "2021-04-26T00:00:00.000000Z"}}}
+
+        customers = list(CUSTOMER_OBJECT.get_objects())
+
+        self.assertNotIn("since_id", Context.state["bookmarks"]["customers"])
+        self.assertNotIn("updated_at_max", Context.state["bookmarks"]["customers"])
+
+    @mock.patch("tap_shopify.streams.base.Stream.call_api")
+    @mock.patch("tap_shopify.streams.base.Stream.get_bookmark", return_value=strptime_to_utc('2021-08-11T01:55:05-04:00'))
+    @mock.patch("tap_shopify.streams.base.Stream.update_bookmark")
+    @mock.patch("tap_shopify.streams.base.Stream.get_query_params")
+    @mock.patch("singer.utils.now", return_value=strptime_to_utc(NOW_TIME))
+    def test_interrupted_sync(self, mock_now, mock_get_query_params, mock_update_bookmark, mock_get_bookmark, mock_call_api):
+        """Verify if sync is interrrupted twice in a row then since_id and updated_at_max keys are not deleted from the state"""
+        mock_call_api.side_effect = Exception("Dummy exception...")
+        Context.state = {"bookmarks": {
+            "currently_sync_stream": 'customers',
+            "customers": {
+                "updated_at": "2021-03-27T00:00:00.000000Z",
+                "since_id": 15,
+                "updated_at_max": "2021-04-26T00:00:00.000000Z"}}}
+
+        try:
+            customers = list(CUSTOMER_OBJECT.get_objects())
+        except:
+            pass
```

Comment:
It feels like we should catch our dummy exception because in the current implementation, if an exception is thrown or not, we handle it the same.

How do we distinguish a syntax error in the test from the dummy exception, for example?

Reply:
Added dummy exception to distinguish dummy exception from code exception and used it as a side effect.